nixos/modules/services/web-apps/rimgo.nix

```diff
@@ -0,0 +1,94 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}: let
+  cfg = config.services.rimgo;
+  inherit (lib)
+    mkOption
+    mkEnableOption
+    mkPackageOption
+    mkDefault
+    mkIf
+    types
+    literalExpression
+  ;
+in {
+  options.services.rimgo = {
+    enable = mkEnableOption "rimgo";
+    package = mkPackageOption pkgs "rimgo" {};
+    settings = mkOption {
+      type = with types; attrsOf str;
+      default = {};
+      example = literalExpression ''
+        {
+          FORCE_WEBP = "1";
+        }
+      '';
+      description = ''
+        Settings for rimgo, see [the official documentation](https://rimgo.codeberg.page/docs/usage/configuration/) for supported options.
+      '';
+    };
+    port = mkOption {
+      type = types.port;
+      default = 3000;
+      example = 69420;
+      description = "The port to use.";
+    };
+    address = mkOption {
+      type = types.str;
+      default = "127.0.0.1";
+      description = "The address to listen on.";
+    };
```

Comment:
Why don't we just set the default for `settings`?

Reply:
I removed the `address` option and just set `"127.0.0.1"` as the default address, but as i have pointed out in the other suggestion above, port is more complex and afaik, usually a separate option, so i left that in there for now.